new/missing-method-missing.md

```diff
@@ -0,0 +1,213 @@
+---
+RFC: unassigned
+Author: John Keiser <john@johnkeiser.com>
+Status: Draft
+Type: Standards Track
+---
+
+# Remove magic and method_missing from Chef::Resource lookup
+
+This proposal:
+
+- Switches to explicit methods for Chef recipe DSL instead of method_missing
+- Removes all special treatment of classes in the Chef::Resource and Chef::Provider namespaces
+- Automatically adds resource DSL for all named descendants of Chef::Resource, no matter what their namespace
+
+## Compatibility
+
+In Chef 12, three behavior changes will be noted:
+
+- Stack traces for resource declarations will usually no longer contain
+  `method_missing` (and will instead contain the name of the resources).
+- Classes outside of the Chef::Resource namespace will now be placed in recipe
+  DSL automatically.
+- Warnings are issued for deprecated behavior that will change in Chef 13 (see
+  Deprecations section).
+
+In Chef 13, deprecated behavior will be removed (along with method_missing).  All
+of these have deprecation warnings enabled for Chef 12 (see Deprecations section).
+
+## Motivation
+
+    As a Chef user,
+    I want resources declared in my own namespace to be in recipe DSL automatically,
+    so that I don't have to write the name of the resource twice.
+
+    As a Chef user,
+    I want to declare resources in my own namespace,
+    So that I get better error messages when there is a resource conflict.
+
+    As a Chef developer,
+    I want meaningful names in stack traces for methods,
+    So that I can actually read the debug output.
+
+    As a Chef developer,
+    I want less if statements, case statements and metaprogramming in DSL method calls,
+    So that I can debug and follow logic more easily.
+
+## Specification
+
+### Create explicit DSL methods for resources
+
+When looking up resources, there is no method anywhere that corresponds to the
+actual resource name.  This can lead to a lot of unnecessarily heavy thinking
+when following some very typical stack traces.  It also means that the name of
+the resource being declared is often not on the stack, making debugging harder.
+
+Here we propose that all resource and definition DSL be added to
+`Chef::DSL::Recipe` as actual methods, as they are created, so that they can be
```

Comment:
The actual solution I implemented was a module named `Chef::DSL::Resources`, which I include into `Chef::DSL::Recipe`.  But I'm unclear what the purpose of a cleanroom style class is here, and whether that solution addresses what you are talking about?  Can you clarify?